meta-mender-core/classes/mender-part-images.bbclass

```diff
@@ -74,14 +74,39 @@ mender_part_image() {
     wks="${WORKDIR}/mender-$suffix.wks"
     rm -f "$wks"
     if [ -n "${IMAGE_BOOTLOADER_FILE}" ]; then
+
+        spl_align_kb=0
+        if [ -n "${SPL_BINARY}" ]; then
+            if [ -z "${MENDER_SPL_OFFSET_KB}" ] ; then
+                bberror "MENDER_SPL_OFFSET unspecified.  Please consult your board" \
+                        "to determine the correct value."
+            fi
+
+            # Copy the files to embed in the WIC image into ${WORKDIR} for exclusive access
+            install -m 0644 "${DEPLOY_DIR_IMAGE}/${SPL_BINARY}" "${WORKDIR}/"
+
+            spl_align_kb=${MENDER_SPL_OFFSET_KB}
+            spl_size=$(stat -c '%s' "${WORKDIR}/${SPL_BINARY}")
+            spl_end=$(expr $spl_align_kb \* 1024 + $spl_size)
+            if [ $spl_end -gt $(expr ${IMAGE_BOOTLOADER_OFFSET} \* 1024) ]; then
+                bberror "Size of SPL specified in SPL_BINARY is too large." \
+                        "It will be overwritten by the IMAGE_BOOTLOADER_FILE." \
+                        "Please raise it manually."
+            fi
+            cat >> "$wks" <<EOF
+# embed SPL
+part --source rawcopy --sourceparams="file=${WORKDIR}/${SPL_BINARY}" --ondisk mmcblk0 --align $spl_align_kb --no-table
+EOF
+        fi
+
         # Copy the files to embed in the WIC image into ${WORKDIR} for exclusive access
         install -m 0644 "${DEPLOY_DIR_IMAGE}/${IMAGE_BOOTLOADER_FILE}" "${WORKDIR}/"
 
         if [ $(expr ${IMAGE_BOOTLOADER_BOOTSECTOR_OFFSET} % 2) -ne 0 ]; then
             bbfatal "IMAGE_BOOTLOADER_BOOTSECTOR_OFFSET must be aligned to kB" \
                     "boundary (an even number)."
         fi
-        bootloader_align_kb=$(expr $(expr ${IMAGE_BOOTLOADER_BOOTSECTOR_OFFSET} \* 512) / 1024)
+        bootloader_align_kb=$(expr $(expr $(expr ${IMAGE_BOOTLOADER_BOOTSECTOR_OFFSET} \* 512) / 1024) + $spl_align_kb)
```

Comment:
This I find a bit unintuitive. `IMAGE_BOOTLOADER_BOOTSECTOR_OFFSET` should be absolute, IMHO.

Also, note that `expr` can take more than two operands, and respects normal operator precedence, so we only need one call here.

Reply:
Hmm.  There was already two calls to expr so I was simply modeling my change on that.  Should I try to clean it up?

Reply:
> I'm not sure IMAGE_BOOTLOADER_BOOTSECTOR_OFFSET is absolute in the case of an SPL type configuration. Does the link I posted above give you enough to comment?

Sort of, but I think the proposal highlighted by @mirzak is a better way to handle this.

Reply:
Agreed